src/main.js

```diff
@@ -188,10 +188,53 @@ function languageCodeIsCompatible(language, languages) {
     return languages.some(l => l === languageCode);
 }
 
+function flatpakSystemDirs() {
+  let env = GLib.getenv('EOS_DISCOVERY_FEED_FLATPAK_SYSTEM_DIRS');
+  if (env)
+      return env.split(':');
+
+  return ['/var/lib/flatpak', '/var/endless-extra/flatpak'];
+}
+
+// We need to look at the flatpak directories and /var/lib explicitly
+// since they are not included in XDG_DATA_DIRS
+function allRelevantDataDirs() {
+  return [...(new Set(GLib.get_system_data_dirs().concat(flatpakSystemDirs().map((directory) =>
+    GLib.build_filenamev([directory, 'exports', 'share'])
+  )).concat(['/var/lib']))).values()];
+}
+
+//
+// flatpakCompatibleDesktopInfo
+//
+// Build a GDesktopAppInfo object for a .desktop file that might
+// be in the exports directory but is not necessarily executable
+// because the binary was not mounted in the bwrap jail.
+function flatpakCompatibleDesktopInfo(desktopId) {
+  for (let directory of allRelevantDataDirs()) {
+    let path = GLib.build_filenamev([directory, 'applications', desktopId]);
+    let keyfile = new GLib.KeyFile();
+
+    try {
+        keyfile.load_from_file(path, GLib.KeyFileFlags.NONE);
+    } catch(e) {
+        continue;
+    }
+
+    // Now that we have the keyfile, set the Exec line to some
+    // well-known binary, so that GDesktopAppInfo doesn't trip up
+    // when we try to read it.
+    keyfile.set_string('Desktop Entry', 'Exec', '/bin/true');
```

Comment:
You can use the `G_KEY_FILE_DESKTOP_GROUP` and `G_KEY_FILE_DESKTOP_KEY_EXEC` defines